### PR TITLE
Enforce rustfmt on CI

### PR DESCRIPTION
**Pull request by pyfisch**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22126)

### CODE REVIEW

**Comment by emilio** [comment]
If we do this, which makes more annoying for me to port Gecko changes, can we enable the reordering of imports and such, and remove the tidy one?

At least that'd be roughly neutral or even positive on the effort it takes me :)

Also, maybe, remove the `match_block_trailing_comma` thing while at it?

File: python/servo/testing_commands.py [hunk 1]
```diff
@@ -304,6 +304,12 @@ def test_content(self):
               "tests/wpt/mozilla/.")
         return 0
 
+    def install_rustfmt(self):
+        if self.call_rustup_run(["cargo", "fmt", "--version", "-q"],
+                                stderr=open(os.devnull, "w")) != 0:
```
Comment (emilio) [inline python/servo/testing_commands.py]: Doesn't this leak the file descriptor?

**Comment by jdm** [comment]
Let's have two more commits - one that disables the tidy check for import ordering, and one that enables rustfmt import reordering and formats the whole tree with it.

**Review by jdm — APPROVED** [review]

File: python/servo/testing_commands.py (hunk 1 above)
Comment (pyfisch) [reply python/servo/testing_commands.py]: Good catch!

**Comment by jdm** [comment]
@bors-servo r-
test-tidy complains about some line lengths now.

**Comment by jdm** [comment]
Also apparently travis is not set up to run cargo fmt yet:
```
./components/script/dom/bluetoothcharacteristicproperties.rs:6: Line is longer than 120 characters
./components/script/dom/bluetoothremotegattcharacteristic.rs:8: Line is longer than 120 characters
./components/script/dom/bluetoothremotegattcharacteristic.rs:10: Line is longer than 120 characters
./components/script/dom/bluetoothremotegattdescriptor.rs:8: Line is longer than 120 characters
Running the dependency licensing lint...
It looks like rustup is not installed. See instructions at https://github.com/servo/servo/#setting-up-your-environment
It looks like rustup is not installed. See instructions at https://github.com/servo/servo/#setting-up-your-environment
It looks like rustup is not installed. See instructions at https://github.com/servo/servo/#setting-up-your-environment
```

**Comment by jdm** [comment]
And the selftests failed:
```
======================================================================
FAIL: test_rust (servo_tidy_tests.test_tidy.CheckTidiness)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/servo/servo/python/tidy/servo_tidy_tests/test_tidy.py", line 108, in test_rust
    self.assertEqual('extra space after {', errors.next()[2])
AssertionError: 'extra space after {' != 'missing space before }'
----------------------------------------------------------------------
Ran 28 tests in 0.030s
FAILED (failures=1)
The command "./mach test-tidy --no-progress --self-test" exited with 1.
```

**Comment by pyfisch** [comment]
I've pushed the two commits as suggested by @jdm.

There are still a few problems with this PR:

* On Travis CI the first build does not install rustup. For this reason `./mach test-tidy` fails now. Install rustup there too?
* I used `#![rustup::skip]`  in two files to stop rustfmt from mangling whitespace (introduce trailing whitespace to block comments) but this causes the error `error[E0658]: non-builtin inner attributes are unstable`. What to do here?
* Rustfmt formats imports in 4 cases such that the lines are longer than 120 characters which causes a tidy error: `use dom::bindings::codegen::Bindings::BluetoothCharacteristicPropertiesBinding::BluetoothCharacteristicPropertiesMethods;`
* > Also, maybe, remove the match_block_trailing_comma thing while at it?
    I think this a good idea. It does not disrupt test-tidy.

**Comment by pyfisch** [comment]
Rebased and fixed. Lets see if it works in Travis.

**Comment by jdm** [comment]
```
tidy reported no errors.
It looks like rustup is not installed. See instructions at https://github.com/servo/servo/#setting-up-your-environment
It looks like rustup is not installed. See instructions at https://github.com/servo/servo/#setting-up-your-environment
It looks like rustup is not installed. See instructions at https://github.com/servo/servo/#setting-up-your-environment
The command "./mach test-tidy --no-progress --all" exited with 1.
```

**Comment by pyfisch** [comment]
Travis now passes.

AppVeyor has some connectivity issues:
>  The remote server returned an error: (503) Server Unavailable. Service Unavailable

**Comment by SimonSapin** [comment]
Nice! @pyfisch, are you subscribed to the [dev-servo](https://lists.mozilla.org/listinfo/dev-servo) mailing list? I think it’d be nice to send an announcement there with:

* A link to this PR
* Format checking is now part of `./mach test-tidy`, so unformatted code will fail CI
* `./mach fmt` does the formatting
* Import ordering has changed (but `mach fmt` will fix it)

If you prefer I can do it for you :)

**Comment by pyfisch** [comment]
Sent an email.

**Pull request merged**